src/meili/map_matcher.cc

```diff
@@ -532,45 +532,59 @@ void MapMatcher::RemoveRedundancies(const std::vector<StateId>& result)
 std::vector<MatchResults>
 MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t k)
 {
+  if (k <= 0) {
+    throw std::invalid_argument("expect k to be positive but got " + std::to_string(k));
+  }
+
   // Reset everything
   Clear();
 
+  std::vector<MatchResults> best_paths;
+
   // Nothing to do
-  if (measurements.empty())
-    return {};
+  if (measurements.empty()) {
+    best_paths.emplace_back(
+        std::vector<MatchResult>{},
+        std::vector<EdgeSegment>{},
+        -1.0);
```

Comment:
@kevinkreiser WDYT about this cost? should I make it `0.f`?

Reply:
yes in my opinion it should be 0.f